langchain/spark_sql.py

```diff
@@ -0,0 +1,111 @@
+from typing import Any, Iterable, List, Optional
+
+from pyspark.sql import DataFrame, Row, SparkSession
```

Comment:
we'll want to do 
```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
  from pyspark.sql import ...
```
since pyspark is an optional dependency. this will allow us to still type hint properly but without actually importing things at the top level

Reply:
and then if there's places we actually call these classes (like line 19) you'll want to try/except ImportError the relevant class

Reply:
Thanks, updated